metadatastore/utils/testing.py

```diff
@@ -1,18 +1,32 @@
 import uuid
 from metadatastore.api import db_connect, db_disconnect
-
+from metadatastore import conf
+from copy import deepcopy
 
 conn = None
-db_name = "mds_testing_disposable_{}".format(str(uuid.uuid4()))
+testing_config = {
+    'database': "mds_testing_disposable_{}".format(str(uuid.uuid4())),
+    'host': 'localhost',
+    'port': 27017,
+    'timezone': 'US/Eastern'}
+
+old_connection_info = None
 
 
 def mds_setup():
     "Create a fresh database with unique (random) name."
     global conn
+    global old_connection_info
+    old_connection_info = deepcopy(conf.connection_config)
+    conf.connection_config = testing_config
     db_disconnect()
-    conn = db_connect(db_name, 'localhost', 27017)
+    conn = db_connect(testing_config['database'], testing_config['host'],
```

Comment:
You can probably drop this and rely on `ensure_connection` to take care of the actual connection for you.